client/components/modals/SelectKeywords.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState, useEffect } from 'react';
+
+import { keywordsSamples } from '../../constants/keywords-sample';
+
+import {
+  Box,
+  Heading,
+  Input,
+  InputGroup,
+  InputLeftAddon,
+  Container,
+  Text,
+  Stack,
+  SimpleGrid,
+  Tag,
+  TagLabel,
+  TagCloseButton,
+  TagRightIcon,
+  ButtonGroup,
+} from '@chakra-ui/react';
+
+import {
+  ButtonBlue,
+  ButtonOrange,
+} from '../../styles/ui-components/Chakra-Button';
+
+import { AddIcon, SearchIcon } from '@chakra-ui/icons';
+
+type KeywordProps = {
+  keywordsData: any;
+  keywordsDataHandler: (keywordData: React.FormEvent) => void;
+  closeKeywordModal: () => void;
+};
+
+export default function KeywordSelect({
+  keywordsDataHandler,
+  closeKeywordModal,
+  keywordsData,
+}: KeywordProps): any {
+  let keywords = keywordsSamples;
+
+  const [searchKeywords, setSearchKeywords] = useState('');
+  const [selectedKeywords, setSelectedKeywords] = useState<any>([]);
+  const [keywordsActive, setKeywordsActive] = useState(false);
+
+  useEffect(() => {
+    if (keywordsData && selectedKeywords.length === 0) {
+      Object.values(keywordsData).forEach((keyword) => {
+        selectKeyword((keyword as any).keyword);
+      });
+    }
+  }, [selectedKeywords, keywordsData]);
+
+  const selectKeyword = (e: string) => {
+    //Limits the number of selected keywords to 10
+    const isInArray = selectedKeywords.find(
+      (element: any) => element.keyword === e
+    );
+    if (isInArray || selectedKeywords.length >= 10) {
+      return;
+    }
+    //Merges the selected keywords with the state
+    setSelectedKeywords((prevKeywords: any) => {
+      const updatedKeywords: any = [...prevKeywords];
+      updatedKeywords.unshift({ keyword: e, id: Math.random().toString() });
+      return updatedKeywords;
+    });
+    setKeywordsActive(true);
+  };
+
+  //Deletes keywords from selected list
+  const deleteKeywords = async (e: any) => {
+    setSelectedKeywords((prevKeywords: any) => {
+      const updatedKeywords: any = prevKeywords.filter(
+        (keyword: any) => keyword.id !== e
+      );
+      return updatedKeywords;
+    });
+  };
+
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    keywordsDataHandler(selectedKeywords);
+    closeKeywordModal();
+  };
+
+  return (
+    <>
+      <Box
```

Comment:
Why are we using an empty box, I think an on-click-outside listener is going to be a lot easier